elements/lisk-p2p/src/peer/base.ts

```diff
@@ -464,6 +469,13 @@ export class Peer extends EventEmitter {
 				this._peerConfig.maxPeerInfoSize,
 			);
 		} catch (error) {
+			if (
+				error instanceof InvalidPeerError ||
+				error instanceof InvalidPeerListError
+			) {
+				this.applyPenalty(this._peerConfig.invalidPeerListPenalty);
```

Comment:
You are banning the peer itself which is requesting the list of peers. We have to ban the sender of the wrong information which comes inside the `response.peerId`. Please, be aware that rectificating this will change test expectations too.